Homework 18

Ula1ri opened the pull request:
Доброго дня,  робив все як вказано в завданні, але чомусь не спрацьовує, допоможіть зрозуміти, що я роблю не так, дякую!

Ula1ri committed:
Homework 18

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,32 @@
+    def __init__(self, name,):
+        self.name = name
+    def say_name(self, name):
+        return Bot(self.say_name)
```
task was to print name here
```suggestion
        print(self.name)
```

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,32 @@
+    def say_name(self, name):
+        return Bot(self.say_name)
+    def send_message(self, message):
+        return Bot(self.send_message())
```
```suggestion
        print(message)
```

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,32 @@
+
+
+class TelegramBot(Bot):
+    def __init__(self, url, chat_id):
```
I think the task was to inherit `init` method
```suggestion
    def __init__(self, name, url=None, chat_id=None):
        super().__init__(name)
```

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,32 @@
+        print(url)
+    def set_chat_id(self, chat_id):
+        self.chat_id = chat_id
+        print(f'TG bot says{message} to chat {chat_id} using {url}')
```
this message should be printed in `send_massage` method
```suggestion
        print(f'TG bot says{message} to chat {self.chat_id} using {self.url}')
```

acman reviewed (changes requested):
Overall looks good but I think there was some misunderstanding about the task.

Ula1ri committed:
some updates

Ula1ri committed:
Some Updates2

Ula1ri commented:
Я вніс указані вище правки, але є ще щось що я зробив не так як треба.

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,33 @@
+class Bot:
+    def __init__(self, name,):
```
```suggestion
    def __init__(self, name):
```

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,33 @@
+        self.chat_id = chat_id
+    def send_massage(self, message):
+        super().send_message()
+        return TelegramBot(self.send_message())
```
here method  `send_message` from `Bot` should be rewritten
```suggestion
        print(f'{self.name} bot says {message} to chat {self._chat_id} using {self._url}')
```

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,33 @@
+        print(url)
+    def set_chat_id(self, chat_id):
+        self.chat_id = chat_id
+        print(f'TG bot says{message} to chat {self.chat_id} using {self.url}')
```
remove this line here, it will be printed in `send_message`
```suggestion
```

acman reviewed (changes requested):
Almost there! I suggest it would be helpful for you to rewatch the video about class inheritance or revise the workbook about this topic and try to practice more.

Ula1ri committed:
update 3

Прошу вибачення за купу помилок, код тепер виконується але є питання. Щоб він спрацьовував я вказую в команді some_bot.say_name() в дужках ім'я классу. В іншому випадку   інтерпритатор видає помилку. Це так має бути, чи я щось роблю не так? Дякую за терпіння!

Ula1ri committed:
update 3

Ula1ri commented:
Прошу вибачення за купу помилок, код тепер виконується але є питання. Щоб він спрацьовував я вказую в команді `some_bot.say_name() `в дужках ім'я классу. В іншому випадку   інтерпритатор видає помилку. Це так має бути, чи я щось роблю не так? Дякую за терпіння!

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,36 @@
+class Bot:
+    def __init__(self, name):
+        self.name = name
+    def say_name(self, name):
```
Aaah! here is the problem, method `say_name` waits `name` as required argument
```suggestion
    def say_name(self):
```

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,36 @@
+
+
+some_bot = Bot('Marvin')
+some_bot.say_name(Bot)
```
```suggestion
some_bot.say_name()
```

acman commented on Lecture_18/Homework_18.py:
```diff
@@ -0,0 +1,36 @@
+some_bot.send_message("Hello")
+
+telegram_bot = TelegramBot("TG")
+telegram_bot.say_name(TelegramBot)
```
```suggestion
telegram_bot.say_name()
```

acman reviewed (changes requested):
and final fixes :)

acman commented:
> Прошу вибачення за купу помилок, код тепер виконується але є питання. Щоб він спрацьовував я вказую в команді `some_bot.say_name() `в дужках ім'я классу. В іншому випадку інтерпритатор видає помилку. Це так має бути, чи я щось роблю не так? Дякую за терпіння!

та все гуд, ви тут, щоб вчитися, треба просто більше практики роботи з класами
метод `say_name` чекає обов'язкового аргумента `name` але його не выкорыстовує, тому туди можна будьщо передати, не тільки ім'я классу. Треба цей аргумент видалити.

Ula1ri committed:
update 4

Ula1ri commented:
Тепер працює корректно, дякую за настанови!
